token/js/src/extensions/transferHook/actions.ts

```diff
@@ -65,3 +75,156 @@ export async function updateTransferHook(
 
     return await sendAndConfirmTransaction(connection, transaction, [payer, ...signers], confirmOptions);
 }
+
+/**
+ * Add extra accounts needed for transfer hook to an instruction
+ *
+ * @param connection      Connection to use
+ * @param instruction     The transferChecked instruction to add accounts to
+ * @param commitment      Commitment to use
+ * @param programId       SPL Token program account
+ *
+ * @return Instruction to add to a transaction
+ */
+export async function addExtraAccountsToInstruction(
+    connection: Connection,
+    instruction: TransactionInstruction,
+    mint: PublicKey,
+    commitment?: Commitment,
+    programId = TOKEN_2022_PROGRAM_ID
+): Promise<TransactionInstruction> {
+    if (!programSupportsExtensions(programId)) {
+        throw new TokenUnsupportedInstructionError();
+    }
+
+    const mintInfo = await getMint(connection, mint, commitment, programId);
+    const transferHook = getTransferHook(mintInfo);
+    if (transferHook == null) {
+        return instruction;
+    }
+
+    const extraAccountsAccount = getExtraAccountMetaAccount(transferHook.programId, mint);
+    const extraAccountsInfo = await connection.getAccountInfo(extraAccountsAccount, commitment);
+    if (extraAccountsInfo == null) {
+        return instruction;
+    }
+
+    const extraAccountMetas = getExtraAccountMetas(extraAccountsInfo);
+
+    const accountMetas = instruction.keys;
+
+    for (const extraAccountMeta of extraAccountMetas) {
+        const accountMeta = resolveExtraAccountMeta(
+            extraAccountMeta,
+            accountMetas,
+            instruction.data,
+            transferHook.programId
+        );
+        accountMetas.push(accountMeta);
+    }
+
+    return new TransactionInstruction({ keys: accountMetas, programId, data: instruction.data });
+}
+
+/**
+ * Transfer tokens from one account to another, asserting the token mint, and decimals
+ *
+ * @param connection     Connection to use
+ * @param payer          Payer of the transaction fees
+ * @param source         Source account
+ * @param mint           Mint for the account
+ * @param destination    Destination account
+ * @param authority      Authority of the source account
+ * @param amount         Number of tokens to transfer
+ * @param decimals       Number of decimals in transfer amount
+ * @param multiSigners   Signing accounts if `owner` is a multisig
+ * @param confirmOptions Options for confirming the transaction
+ * @param programId      SPL Token program account
+ *
+ * @return Signature of the confirmed transaction
+ */
+export async function transferCheckedWithTransferHook(
+    connection: Connection,
+    payer: Signer,
+    source: PublicKey,
+    mint: PublicKey,
+    destination: PublicKey,
+    authority: Signer | PublicKey,
+    amount: bigint,
+    decimals: number,
+    multiSigners: Signer[] = [],
+    confirmOptions?: ConfirmOptions,
+    programId = TOKEN_2022_PROGRAM_ID
```

Comment:
Let's change this to `TOKEN_PROGRAM_ID`, that way we can recommend people to just switch over to use this function instead, and it'll work without any other changes